system-addon/test/schemas/pings.js

```diff
@@ -63,14 +63,34 @@ const SessionPing = Joi.object().keys(Object.assign({}, baseKeys, {
   session_id: baseKeys.session_id.required(),
   page: baseKeys.page.required(),
   session_duration: Joi.number().integer().required(),
-  action: Joi.valid("activity_stream_session").required()
+  action: Joi.valid("activity_stream_session").required(),
+  perf: Joi.object().keys({
+    // Timestamp of the action perceived by the user to trigger the load
+    // of this page.
+    //
+    // Not required at least for error cases where the
+    // observer event doesn't fire
+    load_trigger_ts: Joi.number().positive(),
+    // What was the perceived trigger of the load action?
+    //
+    // Not required at least for error cases where the observer event
+    // doesn't fire
+    load_trigger_type: Joi.valid(["menu_plus_or_keyboard"]),
+    // When the page itself receives an event that document.visibilityState
+    // == visible.
+    //
+    // Not required at least for the (error?) case where the
+    // visibility_event doesn't fire.  (It's not clear whether this
+    // can happen in practice, but if it does, we'd like to know about it).
+    visibility_event_rcvd_ts: Joi.number().positive()
+  }).required()
```

Comment:
I suppose it's not technically required, if none of the child properties are there.  So the answer sort of depends on how we decide the other thing...